hist/histv7/test/binning.cxx

```diff
@@ -118,37 +118,37 @@ TEST(HistImplBinning, EquiDist2D) {
 
    // Here's a visual overview of how binning should work
    //
-   //                    Axis 0
-   //              UF   0.0  1.0  OF
-   //        ------------------------
-   //     A   UF  | 0    1    2    3
-   //     x  -1.0 | 4    5    6    7
-   //     .   0.0 | 8    9    10   11
-   //     1   OF  | 12   13   14   15
+   //                     Axis 0
+   //              UF      0.0    1.0    OF
+   //        --------------------------------
+   //     A   UF  |-1/-9   -10   -11   -5/-12
+   //     x  -1.0 |  -2      1     2     -6
+   //     .   0.0 |  -3      3     4     -7
+   //     1   OF  |-4/-13  -14   -15   -8/-16
```

Comment:
Here is a quick example of how local <-> global bin coordinate conversions can be carried out in this coordinate system, restricted to the 2D case for now: https://gist.github.com/HadrienG2/b21b35319470e2a025d6c3fe8a8792c3 .

I still need to generalize it to N dimensions, but if it works out, would you agree to implement and use this bin indexing convention instead of the one that you went for so far?

Reply:
And now, here's a general version of the algorithm that works for any histogram dimension as long as all axes are non-growable : https://gist.github.com/HadrienG2/b21b35319470e2a025d6c3fe8a8792c3 .

(Growable axes are now mocked by the test harness, but do not work yet. Some code somewhere is likely assuming that axes always have under/overflow bins, one way or another.)

Reply:
Alright, here's a version that I don't manage to break anymore, no matter what axis configuration I put in: https://gist.github.com/HadrienG2/b21b35319470e2a025d6c3fe8a8792c3 .

Would you agree with switching the RHistImpl implementation to those `ComputeGlobalBin` and `ComputeLocalBins` algorithms?

Note that the testing methodology used by this toy code may be something we'd want to reuse in the RHist binning tests. Binning code is super-tricky, so it is best covered by a super-paranoid test suite, accounting several axis binning configurations: growable and non-growable axes, different numbers of bins on each axis, in different orders to make sure that the boundaries of min/max/clamp-based algorithms are written correctly...